src/pyaro/timeseries/Filter.py

```diff
@@ -855,4 +859,128 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+
+        :raises TypeError
+            If conversion isn't possible.
+        """
+        # Convert altitude to meters
+        units = Unit(self._topography[self._topo_var].units)
+        if units.is_convertible(self.UNITS_METER):
+            self._topography[self._topo_var].values = self.UNITS_METER.convert(self._topography[self._topo_var].values, self.UNITS_METER)
+            self._topography[self._topo_var]["units"] = str(self.UNITS_METER)
+        else:
+            raise TypeError(f"Expected altitude units to be convertible to 'm', got '{units}'")
+        
+    def _find_lat_lon_variables(self):
+        """
+        Determines the names of variables which represent the latitude and longitude
+        dimensions in the topography data.
+
+        These are assigned to self._lat, self._lon, respectively for later use. 
+        """
+        for var_name in self._topography.coords:
+            unit_str = self._topography[var_name].attrs.get("units", None)
+            if unit_str in self.UNITS_LAT:
+                self._lat = var_name
+                continue
+            if unit_str in self.UNITS_LON:
+                self._lon = var_name
+                continue
+        
+        if any(x is None for x in [self._lat, self._lon]):
+            raise ValueError(f"Required variable names for lat, lon dimensions could not be found in file '{self._topo_file}")
+    
+    def _gridded_altitude_from_lat_lon(self, lat: float, lon: float) -> float:
+        # TODO: Include a tolerance?
+        data = self._topography.sel({self._lat: lat, self._lon: lon}, method="nearest")
+        
+        # Should not vary in time too much so picking the first one here.
+        altitude = data[self._topo_var][0]
+
+        return float(altitude)
+
+    def _is_close(self, alt_gridded: float, alt_station: float) -> bool:
+        """
+        Function to check if two altitudes are within a relative tolerance of each
+        other.
+        
+        :param alt_gridded : Gridded altitude (in meters).
+        :param alt_station : Observation / station altitude (in meters).
+
+        :returns :
+            True if the absolute difference between alt_gridded and alt_station is
+            <= self._rdiff
+        """
+        return abs(alt_gridded-alt_station) <= self._rdiff
+    
+    def init_kwargs(self):
+        return {
+            "topo_file": self._topo_file,
+            "topo_var": self._topo_var,
+            "rdiff": self._rdiff 
+        }
+
+    def name(self):
+        return "relaltitude"
+
+    def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
+        if self._topography is None:
+            return stations
+        
+        filtered_stations = dict()
+
+        for name, station in stations.items():
+            lat = station["latitude"]
+            lon = station["longitude"]
+
+            altobs = station["altitude"]
+            topo = self._gridded_altitude_from_lat_lon(lat, lon)
```

Comment:
The EEA dataset has 75 000 stations. I'm worried that a loop over the _gridded_altitude_from_lat_lon can be quite slow. Can you test performance with a long stations list and see if you can numpy-fy the access, e.g. by getting topo for all lat/lon points in one operation.